### PR TITLE
Include Planets Being Colonized in Colony Upkeep

### PR DESCRIPTION
**Pull request by geoffthemedio**
Will hopefully give an indication of how much colonization will increase emipre influence upkeep.

Will overestimate costs the turn after if the colonization is actually outposting.

https://www.freeorion.org/forum/viewtopic.php?p=115571#p115571

### CODE REVIEW

**Comment by agrrr3** [comment]
ah nice.

this increases the next turn preview without increasing the current turn value because setting the orders happens after effect evaluation.

Are the orders removed before next turn effect evaluation? Else such colonies are probably counted twice.

Also this does not work for colony buildings, does it?

**Comment by geoffthemedio** [comment]
> Are the orders removed before next turn effect evaluation? Else such colonies are probably counted twice.

Effects are evaluated multiple times per turn, so probably yes and no, but this counts ships ordered to colonize, and the ship is destroyed as part of creating a colony, so that shouldn't be double counted.

> Also this does not work for colony buildings, does it?

It just counts ships. Assuming colony buildings exist for a turn before destroying themselves and populating the planet they are on, they could similarly be counted on the turn they are there.

**Comment by agrrr3** [comment]
> Assuming colony buildings exist for a turn before destroying themselves and populating the planet they are on, they could similarly be counted on the turn they are there.

yes, that would be perfect (i think your assumption holds as long as nobody is scrapping the building)

**Comment by geoffthemedio** [comment]
added non-exobot colony buildings

File: default/scripting/species/common/influence.macros [hunk 1]
```diff
@@ -18,6 +18,23 @@ BASE_INFLUENCE_COSTS
...
                     ] +
+                    Statistic Count condition = And [
+                        Ship
+                        OwnedBy empire = Source.Owner
+                        (LocalCandidate.OrderedColonizePlanetID != -1)
+                    ] +
+                    Statistic Count condition = And [
+                        Building
+                        OwnedBy empire = Source.Owner
+                        Building name = [
```
Comment (o01eg) [inline default/scripting/species/common/influence.macros:29]: Can it use something like tags or so to not require edit this list if new species will be introduced?

Comment (geoffthemedio) [reply default/scripting/species/common/influence.macros:29]: probably yes, though I didn't see any suitable tag already present, and didn't want the extra complication of adding one in the v0.5 version. Colony buildings seem to be currently just tagged with the name of their species, which I could check for separately, but would be no better than just checking the building type name as here...

Comment (geoffthemedio) [reply default/scripting/species/common/influence.macros:29]: doesn't seem to work, though...

edit: fixed it

**Pull request closed without merging**